test/unit/puppet_fact_parser_test.rb

```diff
@@ -233,8 +234,35 @@ def setup
     assert_equal '172.30.43.0', parser.interfaces['local_area_connection_2'][:network]
   end
 
+  test "should create connection between new os and new arch on facts parsing" do
+    assert os_to_arch_connectivity_check
+  end
+
+  test "should create connection between existing os and new arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    assert os_to_arch_connectivity_check
```

Comment:
Ok, so i will do the connection between the two on the Host::Managed.

2015-06-04 17:58 GMT+03:00 Dominic Cleal <notifications@github.com>:

> In test/unit/puppet_fact_parser_test.rb
> <https://github.com/theforeman/foreman/pull/2417#discussion_r31728363>:
>
> > @@ -233,8 +234,35 @@ def setup
> >      assert_equal '172.30.43.0', parser.interfaces['local_area_connection_2'][:network]
> >    end
> >
> > +  test "should create connection between new os and new arch on facts parsing" do
> > +    assert os_to_arch_connectivity_check
> > +  end
> > +
> > +  test "should create connection between existing os and new arch on facts parsing" do
> > +    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
> > +    assert os_to_arch_connectivity_check
>
> I don't think so, since Host::Managed already knows it's going to populate
> both architecture and operating system. It might be preferable than
> carrying this patch in three places.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theforeman/foreman/pull/2417/files#r31728363>.
>